massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -138,14 +156,41 @@ impl SpeculativeLedger {
         // simulate crediting coins to destination address (if any)
         // note that to_addr can be the same as from_addr
         if let Some(to_addr) = to_addr {
-            let new_balance = changes
-                .get_balance_or_else(&to_addr, || self.get_balance(&to_addr))
-                .unwrap_or_default()
-                .checked_add(amount)
-                .ok_or_else(|| {
+            let old_balance = changes.get_balance_or_else(&to_addr, || self.get_balance(&to_addr));
+            if let Some(old_balance) = old_balance {
+                let new_balance = old_balance.checked_add(amount).ok_or_else(|| {
                     ExecutionError::RuntimeError("overflow in to_addr balance".into())
                 })?;
-            changes.set_balance(to_addr, new_balance);
+                changes.set_balance(to_addr, new_balance);
+            } else if from_addr.is_some() {
```

Comment:
Small problem here: if from_addr is none (eg. coin creation) we still need to credit the target address.

In that case:
* if the address exists, just credit it
* if the addr doesn't exist and there are enough coins to pay for the addr creation, create it and credit it with the remainder of coins
* if the addr doesn't exist and there are not enough coins to create it, simply do nothing

Reply:
Why no errors ?